File: src/main/java/lesson3/Robot.java
package lesson3;

class Robot implements Participiant {
    static int maxJumpHeight = 30;
    static int maxRunDistance = 40;

    @Override
    public boolean jump(int height) {
        if (height <= Robot.maxJumpHeight) {
            System.out.println("Робот успешно прыгнул");
            return true;
        }
        System.out.println("Робот не прыгнул");
        return false;
    }

    @Override
    public boolean run(int distance) {
        if (distance <= Robot.maxRunDistance) {
            System.out.println("Робот успешно пробежал");
            return true;
        }
        System.out.println("Робот не пробежал");
        return false;
    }
}

File: src/main/kotlin/lesson3_kt/Robot.kt
package lesson3_kt

class Robot : Participian{
    companion object {
        val ROBOT_MAX_DISTANCE = 30
        val ROBOT_MAX_HEIGHT = 40
    }
    override fun run(distance: Int) : Boolean = performAction(
        value = distance,
        maxValue = ROBOT_MAX_DISTANCE,
        action = "пробежал",
        actor = "робот"
    )

    override fun jump(height: Int) : Boolean= performAction(
        value = height,
        maxValue= ROBOT_MAX_HEIGHT,
        action = "прыгнул",
        actor = "робот"
    )
}